GoogleForCreators/web-stories-wp
# Make trim button functional

## Summary
Adds the missing part of making the trim work.


## Relevant Technical Choices
- Exits edit mode and displays a spinner for the time of Trimming.


## To-do



## User-facing changes
| State | Image |
|----|-----|
| During trimming - design panel | <img width="305" alt="Screenshot 2021-09-24 at 15 41 30" src="https://user-images.githubusercontent.com/3294597/134675974-6acce742-2373-424d-9765-5885be0af633.png"> |
| During trimming - quick action menu (hidden Trim action) | <img width="298" alt="Screenshot 2021-09-24 at 15 41 38" src="https://user-images.githubusercontent.com/3294597/134676086-520c4a14-258e-4b03-8796-8d6042429237.png"> |
| Before trimming - origin video in Media Panel | <img width="257" alt="Screenshot 2021-09-24 at 15 41 45" src="https://user-images.githubusercontent.com/3294597/134676157-4f41a442-2003-42ad-a16a-47fc0eb021fc.png"> |
| After trimming - new video in Media Panel | <img width="179" alt="Screenshot 2021-09-24 at 15 41 56" src="https://user-images.githubusercontent.com/3294597/134676234-8435acea-f74e-418f-95ce-f91b82a032c6.png"> |



## Testing Instructions





This PR can be tested by following these steps:

1. Ensure that video trimming experiment is enabled.
2. Add a video (better to take a video that's a bit longer, e.g. 30 seconds, it will give more time to verify the things happening during trimming).
3. Click on the "Trim" button in the video options to ender Trim mode.
4. Trim the video from both ends using the handles.
5. Click "Trim" button that appeared on the right side of the trim bar.
6. Verify that Trim edit mode is exited.
7. Verify that Trim button in the design panel shows now "Trimming...", a spinner, and is disabled.
8. Verify that Mute button in the design panel below the Trim button is also disabled while trimming.
9. Verify that the "Trim" icon in the element menu (quick actions menu) is not visible while trimming.
10. Once trimming has ended, verify that a new video was added to the Media Library panel with correct length.

## Checklist




---



Fixes #9105
